src/servers/src/otlp/logs.rs

```diff
@@ -355,35 +358,321 @@ fn build_identity_row(
                 .map(|x| ValueData::StringValue(log_body_to_string(x))),
         },
     ];
-
     Row { values: row }
 }
 
-fn parse_export_logs_service_request_to_rows(request: ExportLogsServiceRequest) -> Rows {
-    let mut result = Vec::new();
-    for r in request.resource_logs {
+fn extract_field_from_attr_and_combine_schema(
+    schema_info: &mut SchemaInfo,
+    log_select: &SelectInfo,
+    jsonb: &jsonb::Value<'static>,
+) -> Result<Vec<GreptimeValue>> {
+    if log_select.keys.is_empty() {
+        return Ok(Vec::new());
+    }
+    let mut append_value = Vec::with_capacity(schema_info.schema.len());
+    for _ in schema_info.schema.iter() {
+        append_value.push(GreptimeValue { value_data: None });
+    }
+    for k in &log_select.keys {
+        let index = schema_info.index.get(k).copied();
+        if let Some(value) = jsonb.get_by_name_ignore_case(k).cloned() {
+            if let Some((schema, value)) = decide_column_schema(k, value)? {
+                if let Some(index) = index {
+                    let column_schema = &schema_info.schema[index];
+                    ensure!(
+                        column_schema.datatype == schema.datatype,
+                        IncompatibleSchemaSnafu {
+                            column_name: k.clone(),
+                            datatype: column_schema.datatype().as_str_name(),
+                            expected: column_schema.datatype,
+                            actual: schema.datatype,
+                        }
+                    );
+                    append_value[index] = value;
+                } else {
+                    let key = k.clone();
+                    schema_info.schema.push(schema);
+                    schema_info.index.insert(key, schema_info.schema.len() - 1);
+                    append_value.push(value);
+                }
+            }
+        }
+    }
+    Ok(append_value)
+}
+
+fn decide_column_schema(
+    column_name: &str,
+    value: JsonbValue,
+) -> Result<Option<(ColumnSchema, GreptimeValue)>> {
+    let column_info = match value {
+        JsonbValue::String(s) => Ok(Some((
+            GreptimeValue {
+                value_data: Some(ValueData::StringValue(s.into())),
+            },
+            ColumnDataType::String,
+            SemanticType::Tag,
+            None,
+        ))),
+        JsonbValue::Number(n) => match n {
+            JsonbNumber::Int64(i) => Ok(Some((
+                GreptimeValue {
+                    value_data: Some(ValueData::I64Value(i)),
+                },
+                ColumnDataType::Int64,
+                SemanticType::Tag,
+                None,
+            ))),
+            JsonbNumber::Float64(f) => Ok(Some((
+                GreptimeValue {
+                    value_data: Some(ValueData::F64Value(f)),
+                },
+                ColumnDataType::Float64,
+                SemanticType::Tag,
+                None,
+            ))),
```

Comment:
Maybe we should set it as field if it is float as a float tag is usually meaningless.

Reply:
We only extract columns for tag, so here we can move float type to unsupported data type